Sources/RoktUXHelper/Data/Model/ExperienceResponse/CatalogItem.swift

```diff
@@ -13,13 +13,50 @@ import Foundation
 
 struct CatalogItem: Codable {
     let images: [String: CreativeImage]?
-    let catalogItemId: String
-    let cartItemId: String
-    let instanceGuid: String
-    let title: String
-    let description: String
-    let priceFormatted: String
-    let positiveResponseText: String
-    let negativeResponseText: String
-    let token: String
+    let catalogItemId: String?
+    let cartItemId: String?
+    let instanceGuid: String?
+    let title: String?
+    let description: String?
+    let price: Decimal?
+    let originalPrice: Decimal?
```

Comment:
might risk having floating point [issue](https://github.com/swiftlang/swift-corelibs-foundation/issues/4255).

Reply:
Good catch

Reply:
Thanks for capturing it @WeiYewT .
As we discussed, the better plan is to receive it as String format and then convert it to Decimal. This requires changes to Mobile-API, and we will do it after the POC is finalized, before releasing the alpha version.